external-crates/move/crates/move-command-line-common/src/parser.rs

```diff
@@ -136,18 +142,39 @@ impl<'a, Tok: Token, I: Iterator<Item = (Tok, &'a str)>> Parser<'a, Tok, I> {
 }
 
 impl<'a, I: Iterator<Item = (TypeToken, &'a str)>> Parser<'a, TypeToken, I> {
+    pub fn parse_module_id(&mut self) -> Result<ParsedModuleId> {
+        let (tok, contents) = self.advance_any()?;
+        self.parse_module_id_impl(tok, contents)
+    }
+
     pub fn parse_type(&mut self) -> Result<ParsedType> {
         self.parse_type_impl(0)
     }
 
+    pub fn parse_module_id_impl(
+        &mut self,
+        tok: TypeToken,
+        contents: &'a str,
+    ) -> Result<ParsedModuleId> {
+        let tok = match tok {
+            TypeToken::Ident => ValueToken::Ident,
+            TypeToken::AddressIdent => ValueToken::Number,
+            tok => bail!("unexpected token {tok}, expected address"),
+        };
+        let address = parse_address_impl(tok, contents)?;
+        self.advance(TypeToken::ColonColon)?;
+        let name = self.advance(TypeToken::Ident)?.to_owned();
+        Ok(ParsedModuleId { address, name })
+    }
+
     fn parse_type_impl(&mut self, depth: u64) -> Result<ParsedType> {
         self.count += 1;
 
         if depth > MAX_TYPE_DEPTH || self.count > MAX_TYPE_NODE_COUNT {
             bail!("Type exceeds maximum nesting depth or node count")
         }
-        let (tok, contents) = self.advance_any()?;
-        Ok(match (tok, contents) {
+
+        Ok(match self.advance_any()? {
```

Comment:
this is functionally the same as what we have today right, just sugared up a bit?

Reply:
Yep!